Selecting restroom pin crashes app when gps is not enabled

Crashes on Android 6.0. The stack traces below seem to be part of the problem.

java.lang.RuntimeException: Unable to start activity ComponentInfo{org.refugerestrooms/org.refugerestrooms.views.MainActivity}: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: android.support.v4.app.FragmentManagerState

at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2434)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2494)
at android.app.ActivityThread.access$900(ActivityThread.java:157)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1356)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5551)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:730)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:620)
Caused by: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: android.support.v4.app.FragmentManagerState
at android.os.Parcel.readParcelableCreator(Parcel.java:2432)
at android.os.Parcel.readParcelable(Parcel.java:2358)
at android.os.Parcel.readValue(Parcel.java:2264)
at android.os.Parcel.readArrayMapInternal(Parcel.java:2614)
at android.os.BaseBundle.unparcel(BaseBundle.java:221)
at android.os.Bundle.getBundle(Bundle.java:763)
at com.google.android.gms.maps.internal.ea.a(:com.google.android.gms.DynamiteModulesB:168)
at com.google.maps.api.android.lib6.impl.az.a(:com.google.android.gms.DynamiteModulesB:690)
at com.google.maps.api.android.lib6.impl.da.a(:com.google.android.gms.DynamiteModulesB:81)
at com.google.android.gms.maps.internal.z.onTransact(:com.google.android.gms.DynamiteModulesB:66)
at android.os.Binder.transact(Binder.java:387)
at com.google.android.gms.maps.internal.IMapViewDelegate$zza$zza.onCreate(Unknown Source)
at com.google.android.gms.maps.MapView$zza.onCreate(Unknown Source)
at com.google.android.gms.dynamic.zza$3.zzb(Unknown Source)
at com.google.android.gms.dynamic.zza$1.zza(Unknown Source)
at com.google.android.gms.maps.MapView$zzb.zzvu(Unknown Source)
at com.google.android.gms.maps.MapView$zzb.zza(Unknown Source)
at com.google.android.gms.dynamic.zza.zza(Unknown Source)
at com.google.android.gms.dynamic.zza.onCreate(Unknown Source)
at com.google.android.gms.maps.MapView.onCreate(Unknown Source)
at org.refugerestrooms.views.MainActivity.onCreate(MainActivity.java:420)
at android.app.Activity.performCreate(Activity.java:6272)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2387)

--- a/app/src/main/java/org/refugerestrooms/views/MainActivity.java
+++ b/app/src/main/java/org/refugerestrooms/views/MainActivity.java
@@ -828,13 +828,15 @@ public class MainActivity extends ActionBarActivity
                 //mMap.getUiSettings().setMapToolbarEnabled(true);
                 if (mCurrentLocation != null) {
                     navigateToMarker(marker);
-                } else {
-                    mLastLocation = LocationServices.FusedLocationApi.getLastLocation(
-                            mGoogleApiClient);
-                    if (mLastLocation != null) {
-                        navigateToMarker(marker);
-                    }
                 }
+                // was causing java.lang.IllegalArgumentException: GoogleApiClient parameter is required.
+//                else {
+//                    mLastLocation = LocationServices.FusedLocationApi.getLastLocation(
+//                            mGoogleApiClient);
+//                    if (mLastLocation != null) {
+//                        navigateToMarker(marker);
+//                    }
+//                }
                 return true;
             }
         });
@@ -940,19 +942,20 @@ public class MainActivity extends ActionBarActivity
             routing.registerListener(this);
             routing.execute(start, end);
         }
-        else if (mLastLocation != null) {
-            end = marker.getPosition();
-            setActionBarTitle(marker.getTitle());
-
-            double myLat = mLastLocation.getLatitude();
-            double myLng = mLastLocation.getLongitude();
-            start = new LatLng(myLat,myLng);
-            mLocationTitle = marker.getTitle();
-
-            Routing routing = new Routing(Routing.TravelMode.WALKING);
-            routing.registerListener(this);
-            routing.execute(start, end);
-        }
+        // was causing java.lang.IllegalArgumentException: GoogleApiClient parameter is required.
+//        else if (mLastLocation != null) {
+//            end = marker.getPosition();
+//            setActionBarTitle(marker.getTitle());
+//
+//            double myLat = mLastLocation.getLatitude();
+//            double myLng = mLastLocation.getLongitude();
+//            start = new LatLng(myLat,myLng);
+//            mLocationTitle = marker.getTitle();
+//
+//            Routing routing = new Routing(Routing.TravelMode.WALKING);
+//            routing.registerListener(this);
+//            routing.execute(start, end);
+//        }
     }
 
     //TODO Possibly fix navigation drawer to be a smoother switch between the map and add bathroom fragment

